unit_test/test_spdm_crypt/test_spdm_crypt.c

```diff
@@ -401,6 +507,36 @@ void libspdm_test_crypt_spdm_x509_certificate_check(void **state)
         assert_true(status);
         free(file_buffer);
     }
+
+    uint8_t buffer[0x1200];
+    size_t buffer_size;
+    uint8_t data;
+
+    if ((LIBSPDM_RSA_SSA_2048_SUPPORT) && (LIBSPDM_SHA256_SUPPORT)) {
+        libspdm_copy_mem(buffer, m_cert_buffer_size, m_cert_buffer, m_cert_buffer_size);
+        buffer_size = m_cert_buffer_size;
+
+        /* Correct Certificate.cert.der */
+        status = libspdm_x509_certificate_check(buffer, buffer_size,
+                                                SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_2048,
+                                                SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_256,
+                                                true, true);
+        assert_true(status);
+
+        /* The definite form of length encoding shall be used, encoded in the minimum number of octets
+         * Use non canonical encoding format , length encoding is : 0x81 0x03*/
+        data = 0x81;
+        insert_data(buffer, buffer_size, &data, sizeof(data), 9);
+        buffer_size += sizeof(data);
+        buffer[3] += sizeof(data);
+        buffer[7] += sizeof(data);
```

Comment:
what is buffer[3] and buffer[7]?

Reply:
this is of Certificate SEQUENCE  length and tbsCertificate SEQUENCE length